src/hooks/useImmediateEffect.ts

```diff
@@ -0,0 +1,31 @@
+import { useRef, useEffect, useCallback } from 'react';
+
+enum EffectState {
+  BeforeMount = 0,
+  AfterMount = 1,
+  Render = 2,
+}
+
+type Effect = () => void | (() => void);
+
+/** This executes an effect immediately on initial render and then treats it as a normal effect */
+export const useImmediateEffect = (effect: Effect, changes?: any[]) => {
+  const state = useRef(EffectState.BeforeMount);
```

Comment:
Could we rename this to for example lifecycleState or something? Don't want to nit but this could be confusing for newer contributors

Reply:
Lifecycle is a good call 👌